compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -152,9 +152,7 @@ pub fn unsized_info<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
         (
             &ty::Dynamic(ref data_a, _, src_dyn_kind),
             &ty::Dynamic(ref data_b, _, target_dyn_kind),
-        ) => {
-            assert_eq!(src_dyn_kind, target_dyn_kind);
-
+        ) if src_dyn_kind == target_dyn_kind => {
```

Comment:
Note to self: same change needs to be made to cg_clif

Reply:
And also in Miri.